### PR TITLE
Adds integration pages for third-party authentication

### PR DESCRIPTION
**Pull request by hugobarauna**
Adds integration pages for:

- Cloudflare Zero Trust
- Google IAP
- Tailscale

Before we publish this to production, there are some dependencies and other tasks we'd need to handle:

- decide and implement the image previews for each of those new integrations
- publish the docs for those integrations to the upcoming ExDoc of Livebook
- release v0.11 (which will contain those new integrations)

## The pre-v0.11-release branch

This PR is targeting the `pre-v0.11-release` branch instead of the main one, so we can have a branch with everything that needs to be merged to the main once we release v0.11.

## Regarding the image previews for each of those new integrations

Every integration page we have has an image preview:

CleanShot 2023-09-22 at 15 31 28

For those new integrations, I could not think of an image preview, because there's no UI in Livebook for that integration; everything is handled in the backend.

So, for now, I just left a placeholder there. For example:

CleanShot 2023-09-22 at 15 34 53

I need help to decide how to handle that.

Here are a couple of ideas I had:

- ask Aakash to come up with a different design for an integration page that doesn't have a preview image
- ask Aakash to come up with an image to each for each of those new integrations

### CODE REVIEW

**Comment by josevalim** [comment]
One idea for the image is to show the “login” screen these services provide. But they are annoying to obtain.

**Comment by hugobarauna** [comment]
> One idea for the image is to show the “login” screen these services provide. But they are annoying to obtain.

I thought about that, but I was not sure if they were visually appealing enough, or somewhat visually representative of the integration.

Here's the authentication flow using Cloudflare, for example:

https://github.com/livebook-dev/livebook.dev/assets/2719/80cda9b0-4ef1-459f-af02-b9b14ebe3ca3

**Comment by josevalim** [comment]
Also, we may want to post those integrations under the teams page.

**Comment by josevalim** [comment]
Perhaps we could do a montage where their authentication takes half of the image and our Livebook app home takes the other half?

**Comment by josevalim** [comment]
@wojtekmach can help with the tailscale picture if we go down this road.

**Comment by josevalim** [comment]
Also, could we add some sort of visual indication this is related to Livebook Teams? I can also do that later if you can’t. :)

**Comment by hugobarauna** [comment]
I think I'll need someone else to generate those images. My design skills look pretty inexistent. It took me almost one hour to create that image (for cloudflare + livebook), for example:

CleanShot 2023-10-05 at 11 38 18

And I couldn't create what I had in mind, something with a layout like this:

CleanShot 2023-10-05 at 11 40 05

@josevalim should we ask Aakash's help in creating those 3 images? Or would someone else like to give it a try before we ask for his help?

**Comment by josevalim** [comment]
Can you provide the images from the services? Then either Aakash or Paulo can help!

**Comment by hugobarauna** [comment]
@josevalim here the screenshots of the flow of authenticating with Tailscale and Cloudflare. I didn't do for Google IAP yet, because it takes more work, and I wanted to discuss if we'll move forward with that screenshot-based approach before.

## For Cloudflare

### Step 1

When I go the URL of my Cloudflare team, with the list of apps I have access to through cloudflare authentication, then Cloudflare asks me to log in:

cloudflare 1

### Step 2

I click the login button, and Cloudflare will ask me to login with the configured identity provider. In that example, Github.

cloudflare 2

### Step 3

After I login with my identity provider, clodflare gives me the list of apps I can access using cloudflare ZTA, in that case, the Livebook instance is one of the apps:

cloudflare 3

## For Tailscale

## Step 1

Given I already have Tailscale installed on my machine, when I access the URL of the Livebook running inside out Tailnet (Taiscale VPN), then it just works and Livebook gets the name and email of the user authenticated in Tailscale:

tailscale 5

## Possible steps before step 1

When I'm first installing Tailscale and logging in the service, this is the flow:

tailscale 1
tailscale 2
tailscale 3
tailscale 4

# My considerations

Given the screenshots we can take of the authentication process, I'm not sure we can come up with a montage or something else using those screenshots that will be visually appealing for the integration page.

Maybe what we can do is something these uses a screenshot of Livebook, the logo of the integration (Tailscale, Cloudflare, Google IAP), and some other element that indicates that we're using those for authentication.

Here are a few ideas from ChatGPT:

- Cloudflare Zero Trust: Half the image shows Cloudflare's recognizable orange cloud logo or UI, the other half shows the Livebook app home screen. Blend them together in the middle with a lock symbol or an arrow pointing from Cloudflare to Livebook.
- Google Identity-Aware Proxy: Google’s 'G' or the IAP UI on one side, Livebook on the other. Again, blend in the middle, maybe with a Google-colored shield or arrow to indicate secure connection.
- Tailscale: Feature Tailscale's network diagram or logo on one side, Livebook on the other. Use a network-like design element to blend the two halves.

@josevalim do you feel more inclined in one way or another?

I'm open to either path, I just not being able to visualize how we can use the screenshots.

**Comment by hugobarauna** [comment]
@josevalim as we talked, here's the new direction:

## Option 1

option 1

## Option 2

Same as option 1, but reducing the size of the logo

option 2

## Option 3

Logo with some background behind it.

opcao 3


# My considerations

That approach may result in something we can use. This idea of having a screenshot of Livebook in the background and the logo of the authentication provider on top of it tries to communicate that Livebook is being "protected" (authenticated) by the identity provider represented by the logo.

That said, what I could come up with is still not visually pretty. Now, maybe we can ask for some help from Paulo or Aakash to try to create an image with that direction.

@josevalim What do you think?

**Comment by josevalim** [comment]
@hugobarauna I pushed images for all 3 directly to your branch. I have also pushed a template we can use for future additions. Not I have not resized nor cut the images, currently they are bigger than they need to be, feel free to cut+optimize+resize.

**Comment by josevalim** [comment]
Currently it would look like this:

image

File: src/data/integrations.js [hunk 1]
```diff
@@ -387,6 +394,51 @@ const integrations = [
...
     image: mermaidPreview,
   },
+  {
+    id: "cloudflare",
+    name: "Cloudflare Zero Trust",
+    logo: cloudflareLogo,
+    category: "Authentication",
+    headline: "Authenticate with Cloudflare Zero Trust",
+    description:
+      "Use Cloudflare Zero Trust to ensure that only authorized users can access your Livebook instance.",
```
Comment (josevalim) [inline src/data/integrations.js:404]:
```suggestion
      "Use Cloudflare Zero Trust to ensure that only authorized users can access your Livebook instance and deployed apps.",
```

File: src/data/integrations.js [hunk 2]
```diff
@@ -387,6 +394,51 @@ const integrations = [
...
+    },
+  },
+  {
+    id: "google-iap",
+    name: "Google IAP",
+    logo: googleIapLogo,
+    category: "Authentication",
+    headline: "Authenticate with Google Identity-Aware Proxy",
+    description:
+      "Use Google Identity-Aware Proxy (IAP) to ensure that only authorized users can access your Livebook instance.",
```
Comment (josevalim) [inline src/data/integrations.js:419]:
```suggestion
      "Use Google Identity-Aware Proxy (IAP) to ensure that only authorized users can access your Livebook instance and deployed apps.",
```

File: src/data/integrations.js [hunk 3]
```diff
@@ -387,6 +394,51 @@ const integrations = [
...
+    },
+  },
+  {
+    id: "tailscale",
+    name: "Tailscale",
+    logo: tailscaleLogo,
+    category: "Authentication",
+    headline: "Authenticate with Tailscale",
+    description:
+      "Use Tailscale to ensure that only authorized users can access your Livebook instance.",
```
Comment (josevalim) [inline src/data/integrations.js:434]:
```suggestion
      "Use Tailscale to ensure that only authorized users can access your Livebook instance and deployed apps.",
```

**Comment by josevalim** [comment]
As per our call:

1. We will add a section talking about LT: "With Livebook Teams, you can deploy notebooks as applications with pre-configured Zero Trust Authentication"

2. We will sort the sidebar alphabetically

**Comment by hugobarauna** [comment]
@josevalim I think now we're good to go.

Please check, and if ok, you can already merge and deploy. 🚀

**Pull request merged**